gdc_ng_models/models/study.py

```diff
@@ -0,0 +1,134 @@
+"""Data models to describe studies.
+
+Studies will be used for the single-study controlled-access feature in gdcapi.
+These models provide a way to associate programs and projects to studies.
+"""
+
+from sqlalchemy.ext import declarative
+from sqlalchemy.sql import schema, sqltypes
+
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class Study(Base, audit.AuditColumnsMixin):
+    """A study for use with single-study controlled-access.
+
+    Attributes:
+        id: A unique identifier for the study.
+        name: A unique, informational name for the study.
+        created_date: The date and time when the record is created.
+        updated_date: The date and time when the record is updated.
+    """
+
+    __tablename__ = "study"
+    id_seq = schema.Sequence(name="study_id_seq", metadata=Base.metadata)
+    id = schema.Column(sqltypes.Integer, nullable=False, server_default=id_seq.next_value())
```

Comment:
I prefer `primary_key=True` and `ForeignKey` definitions here. imo it's more readable

Reply:
The primary reason is that I wanted to maintain control over the name of the index, instead of handing control to the database. This is beneficial for a few reasons.  First, I believe that we should explicitly control our namespace inside of a database.  Second, it allows someone to see an index in the db, then search for the name in the code.  Third, it allows me to write reliable tests to verify that the data model enforces the correct integrity rules.

For `primary_key=True`, I find it too easy to miss a column when the primary key is natural key of multiple columns.  This is subjective though.

Reply:
I don't have a strong preference, but default names seem to be ok too, e.g. from `qc_test_run` and `qc_validation_result` tables:
```
...
Indexes:
    "qc_test_runs_pkey" PRIMARY KEY, btree (id)
...
Indexes:
    "qc_validation_results_pkey" PRIMARY KEY, btree (id, test_run_id)
```

Reply:
yes, pg has a good default naming scheme, but it's still a good practice to name things instead of relying on the defaults of your db.

Not sure where those `ix_*` names came from.  They do not follow pg's default naming scheme.

Reply:
it might be coming from the sqlalchemy.